### ISSUE TITLE
Transparency Trouble with ImageOverlay and OpenCV

### ISSUE THREAD

**Issue by proofconstruction** [sparkfish/augraphy#92]
I need some help - originally I forgot the most important part of ImageOverlay, which is making sure the overlay becomes transparent before writing it onto the document. This is [fixed now](https://github.com/sparkfish/augraphy/commit/08229bbce240db4cd24cd12325c70cadfffce00b#diff-a9a22b2a5e5b8e29e2a96d0952f1010a321ee69a4b1526fad6c22bec2839faeb), but I'm still having trouble with OpenCV. I *think* I'm just making mistakes working with the alpha channel and keeping the right `np.dtype` for numbers in the image, but I'm not sure.

**Note:** I moved the `make_white_transparent` function out of `AugraphyPipeline` and into `lib`, since we'll need it for these overlays in the future.

---
Here's the problem.

Using this input image,
![clean](https://user-images.githubusercontent.com/74747193/133886891-a9c75a3a-03c3-4afb-815f-8271550669cd.png)

if I use the following code, I get the result below. This is the behavior I wanted; remove the white background from the input image (which is the overlay, in this case), and place it over the background (the random noise).

```python
import numpy as np
import random
import pytest
import cv2

from augraphy import *
from augraphy.augmentations.lib import make_white_transparent

img = cv2.imread('1 - Letter - Diahann Carroll - 1972 - clean.png')

img = make_white_transparent(img)

ink = []

paper = []

post = [
    ImageOverlay(img, position=(0,0), layer="post", p=1)
]

random = np.random.randint(low=0, high=255, size=img.shape, dtype=np.uint8)

pipeline = AugraphyPipeline(ink,paper,post)

complete = pipeline.augment(random)
```

![complete](https://user-images.githubusercontent.com/74747193/133886928-56ddce56-37d4-4391-9c46-d73656a9e41d.png)

However, when I try to use OverlayBuilder to create overlays, they come out all wrong. For example, when I try to make an overlay with this image:
![binder-ring](https://user-images.githubusercontent.com/74747193/133887299-0c6ea37d-a5b1-4cfc-b10b-c6ac7d07ccf1.png)

and this code:

```python
bg = np.ones((1500,1000,3))
fg1 = cv2.imread('binderring.png')
ob1 = OverlayBuilder(fg1, bg, 30, "right", 10)
overlay1 = ob1.buildOverlay()
```

I correctly get this result, if I view it with `cv2.imshow`:
![overlay](https://user-images.githubusercontent.com/74747193/133887329-be533995-85a0-4a11-ad04-7f900d209a9d.png)

When I use this as the overlay and try to put it on the original image above, I get something like this:

![overlay1](https://user-images.githubusercontent.com/74747193/133887344-eae378dc-508f-4835-a547-f863ca6874c6.png)

Also using `cv2.imwrite` with overlays that appear correct (with a white background) makes them show up with a black background like this picture.

**Comment by proofconstruction** [comment]
Once this is sorted, we can use OverlayBuilder to produce an overlay with binder rings on the right side and another one with binder clips on the bottom, then we can use two ImageOverlays (with working transparency) to add the effect seen here

![2 - News - Let Us Have Facts - 1962 - dirty](https://user-images.githubusercontent.com/74747193/133887482-3c8710c3-a3e6-4fa9-8c33-972434433215.png)

**Comment by kwcckw** [comment]
Okay, I will check and test again specifically on this tomorrow , I will let you know again if I get anything useful.

**Comment by kwcckw** [comment]
I checked and it is due to the wrong assignment of background:

     overlayBase = np.ones((ydim, xdim, 3))
     
The base should be background instead of newly initialized variable.

I also added some code to select a darker pixel between foreground and background, so it would be like this:

example1:
![image](https://user-images.githubusercontent.com/52521979/133923811-82827239-09c9-4b6d-b60b-f4076dec9061.png)

example2:
![image](https://user-images.githubusercontent.com/52521979/133923806-358f189a-de9d-4566-864a-d7d8b4e93f6e.png)

Please check my latest pull request again on the changes.

**Comment by proofconstruction** [comment]
Resolved by PR #93
